hooks/post_gen_project.py

```diff
@@ -4,6 +4,8 @@
 import string
 from pathlib import Path
 
+from tomlkit import dumps, parse
```

Comment:
Oh I just realised that this is a 3rd party package, NOT the `tomllib` module from the standard library (which only focuses on parsing TOML). 

Adding a 3rd party package means that we need to explain to our users that they need to `pip install` / `pipx inject` / `uv tool ... --with` that package before they run the cookiecutter CLI: AFAIK cookiecutter does NOT install the template dependencies before generating it.

That's a significant change in usability for this template, not sure we want to start doing that. 

(that's partly why we're doing some pretty basic YAML parsing in the `remove_prettier_pre_commit` instead of using `pyyaml`, although I just realised that `pyyaml` is a dependency of cookiecutter, so we could probably use it)

Reply:
That is a significant drawback so I will revert the changes regarding the manipulation of pyproject.toml and use the `uv add -r` hooks.